src/main/java/rssfeed/RssItem.java

```diff
@@ -0,0 +1,44 @@
+package rssfeed;
+
+import java.util.Date;
+import java.util.Objects;
+
+public class RssItem {
+    public String title;
+    public String link;
+    public String description;
+    public String imgUrl;
+    public Date pubDate = new Date(0);
+
+    public String guid;
+
+    public RssItem() {}
+
+    public RssItem(RssItem other) {
+        title = other.title;
+        link = other.link;
+        description = other.description;
+        imgUrl = other.imgUrl;
+        pubDate.setTime(other.pubDate.getTime());
+
+        guid = other.guid;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass() || this.hashCode() != o.hashCode()) return false;
```

Comment:
```suggestion
        if (!(o instanceof RssItem)) return false;
```
"It is not required that if two objects are unequal according to the equals(java.lang.Object) method, then calling the hashCode method on each of the two objects must produce distinct integer results."